lstchain/scripts/lstchain_dvr_pixselector.py

```diff
@@ -0,0 +1,516 @@
+
+    # Value will be modified, if a too high average fraction (
+    # > max_pix_survival_fraction) of pixels (and FIRST neighbors)
+    # survive, in shower events, with that picture threshold. The value is
+    # computed excluding the 10% noisiest pixels in the camera, because we
+    # want it to be stable when e.g. stars get in and out of the camera,
+    # or when changing wobbles (star field changes)
+    max_pix_survival_fraction = 0.1
+
+
+    # Number of "rings" of pixels to be finally kept around pixels which are
+    # above min_charge_for_certain_selection
+    number_of_rings = args.number_of_rings
+    summary_info.number_of_rings = number_of_rings
+
+    current_run_number = -1
+
+    for dl1_file in dl1_files:
+        print()
+        print('Input file:', dl1_file)
+
+        run_info = parse_dl1_filename(dl1_file)
+        run_id, subrun_id = run_info.run, run_info.subrun
+        summary_info.run_id = run_id
+        summary_info.subrun_id = subrun_id
+
+        output_dir = args.output_dir.absolute()
+        output_dir.mkdir(exist_ok=True, parents=True)
+
+        # The file DVR_settings_LST-1.Run*.h5 will be needed in case the
+        # --action create_pixel_masks  option is used:
+        input_dvr_settings_file = Path(output_dir,
+                                       f'DVR_settings_LST-1.Run{run_id:05d}.h5')
+        dvr_settings = None
+
+        # If we are just calculating the Data Volume Reduction settings, we
+        # write a single file for the whole run:
+        output_file = Path(output_dir, f'DVR_settings_LST-1.Run{run_id:05d}.h5')
+
+        # On the other hand, for writing the pixels masks we will create
+        # subrun-wise files:
+        if write_pixel_masks:
+            output_file = Path(output_dir, f'Pixel_selection_LST-1.Run'
+                                           f'{run_id:05d}.{subrun_id:04d}.h5')
+
+            if not os.path.isfile(input_dvr_settings_file):
+                print('ERROR: --action create_pixel_masks selected, but file ' +
+                      input_dvr_settings_file.name + 'does not exist!')
+                print('You must run first this script over all the subruns of '
+                      'the run in one go, using --action compute_dvr_settings')
+                exit(1)
+
+            dvr_settings = read_table(input_dvr_settings_file, '/run_summary')
+            # We just take the values below from the first table row, because by
+            # construction they are all identical. These are the settings
+            # used in the execution that resulted in the creation of the
+            # DVR_settings_LST-1.Run*.h5 file:
+            summary_info.number_of_rings = dvr_settings['number_of_rings'][0]
+            summary_info.picture_threshold = dvr_settings['picture_threshold'][0]
+
+        print('Output file:', output_file)
+
+        data_parameters = read_table(dl1_file,
+                                     '/dl1/event/telescope/parameters/LST_LSTCam')
```

Comment:
this is the `dl1_params_lstcam_key`
https://github.com/cta-observatory/cta-lstchain/blob/e41d81d1041f9e700d5190df095d7415cd29dd60/lstchain/io/io.py#L79